### PR TITLE
Finish ComplexType

### PR DESCRIPTION
**Pull request by weitzhandler**
Fix another issue with LoadRelatedEntities.
Fixes #185 but now using a new branch to avoid conflicts.

Exception explained in detail in [this](https://github.com/TrackableEntities/trackable-entities/pull/182#issuecomment-317878366) comment.

### CODE REVIEW

File: Source/TrackableEntities.EF.5/DbContextExtensions.cs [hunk 1]
```diff
@@ -391,9 +391,8 @@ where typeof(ITrackable).IsAssignableFrom(prop.PropertyType)
             // Recursively populate related entities on ref and child properties
             foreach (var item in items)
             {
-                // Avoid endless recursion
-                if (!visitationHelper.TryVisit(item)) continue;
-
+                // Avoid loading related entities of complext types, and avoid endless recursion
+                if (!visitationHelper.TryVisit(item) || IsComplexType(context, item.GetType())) continue;
```
Comment (tonysneed) [inline Source/TrackableEntities.EF.5/DbContextExtensions.cs]: You need to make this change also to the async version, `LoadRelatedEntitiesAsync`.

**Comment by weitzhandler** [comment]
@tonysneed 
> This PR need a commit with a failing test.

Ok added a failing test and fixed it.

File: Source/TrackableEntities.EF.5/DbContextExtensions.cs [hunk 2]
```diff
@@ -392,7 +392,7 @@ where typeof(ITrackable).IsAssignableFrom(prop.PropertyType)
             foreach (var item in items)
             {
                 // Avoid loading related entities of complext types, and avoid endless recursion
-                if (!visitationHelper.TryVisit(item) || IsComplexType(context, item.GetType())) continue;
+                if (IsComplexType(context, item.GetType()) || !visitationHelper.TryVisit(item)) continue;
```
Comment (tonysneed) [inline Source/TrackableEntities.EF.5/DbContextExtensions.cs]: No problem for this PR, but in the future try to add the failing test in one comment, then add the fix to another commit.  That way, I can do a soft reset to the commit with the failing test to see it fail, and then reset to the fit commit to see it pass.  Just a bit easier that way.  But don't worry about it this time.

File: Source/Tests/TrackableEntities.EF.5.Tests/LoadRelatedEntitiesTests.cs [hunk 3]
```diff
@@ -829,6 +830,27 @@ public void LoadRelatedEntities_Should_Populate_Multiple_Employees_Territories_W
...
+        public void LoadRelatedEntities_With_Complex_Types()
+        {
+            var context = TestsHelper.CreateFamilyDbContext(CreateDbOptions.DropCreateDatabaseIfModelChanges);
+            var parent = new Parent(Guid.NewGuid().ToString());
+            parent.Address = new Address();
+            context.Parents.Add(parent);
+            context.SaveChanges();
+
+            // Act
+            context.LoadRelatedEntities(parent);
```
Comment (tonysneed) [inline Source/Tests/TrackableEntities.EF.5.Tests/LoadRelatedEntitiesTests.cs]: We'll also need a test for `LoadRelatedEntitiesAsync`.  And of course the fix for that in a separate commit.

Comment (weitzhandler) [inline Source/Tests/TrackableEntities.EF.5.Tests/LoadRelatedEntitiesTests.cs]: OK great, I'll now commit a failing test for the async, then a fix for it with the following commit.

**Comment by weitzhandler** [comment]
Ok.
This time I committed the failing test and the fix in separate commits.
See above.

**Pull request merged**